src/__test__/server.test.js

```diff
@@ -0,0 +1,99 @@
+  beforeAll(startServer); // Vinicio - we don't use startServer() because we need a function
+  afterAll(stopServer);
+  // afterEach(() => User.remove({}));
+  test('POST - It should respond with a 200 status ', () => {
+    const userToPost = {
+      name: faker.name.findName(),
+      location: faker.address.city(),
+      bicepsRPM: faker.random.number(300),
+      tricepsRPM: faker.random.number(300),
+    };
+    return superagent.post(apiURL)
+      .send(userToPost)
+      .then((response) => {
+        // Vinicio - testing status code
+        expect(response.status).toEqual(200);
+        // Vinicio - Testing for specific values
+        expect(response.body.name).toEqual(userToPost.name);
+        expect(response.body.location).toEqual(userToPost.location);
+        expect(response.body.bicepsRPM).toEqual(userToPost.bicepsRPM);
+        expect(response.body.tricepsRPM).toEqual(userToPost.tricepsRPM);
+        // Vinicio - Testing that properties are present
+        expect(response.body._id).toBeTruthy();
+      });
+  });
+  test('POST - It should respond with a 400 status ', () => {
+    const UserToPost = {
+      location: faker.address.city(),
+      bicepsRPM: faker.random.number(300),
+      tricepsRPM: faker.random.number(300),
+    };
+    return superagent.post(apiURL)
+      .send(UserToPost)
+      .then(Promise.reject) // Vinicio - this is needed because we are testing for failures
+      .catch((response) => {
+        // Vinicio - testing status code
+        expect(response.status).toEqual(400);
+      });
+  });
+  describe('GET /api/users', () => {
+    test('should respond with 200 if there are no errors', () => {
+      let userToTest = null; //  Vinicio - we need to preserve the User because of scope rules
+      return createUserMock() // Vinicio - test only a GET request
+        .then((user) => {
+          userToTest = user;
+          return superagent.get(`${apiURL}/${user._id}`);
+        })
+        .then((response) => {
+          expect(response.status).toEqual(200);
+          expect(response.body.name).toEqual(userToTest.name);
+          expect(response.body.location).toEqual(userToTest.location);
+          expect(response.body.bicepsRPM).toEqual(userToTest.bicepsRPM);
+          expect(response.body.tricepsRPM).toEqual(userToTest.tricepsRPM);
+        });
+    });
+    test('should respond with 404 if there is no user to be found', () => {
+      return superagent.get(`${apiURL}/THisIsAnInvalidId`)
+        .then(Promise.reject) // Vinicio - testing for a failure
+        .catch((response) => {
+          expect(response.status).toEqual(404);
+        });
+    });
+  });
+  test('should respond with an array.', () => {
+    return superagent.get(apiURL)
+      .then((response) => {
+        expect(Array.isArray(response.body)).toBeTruthy();
+      });
+  });
+  test('should respond with 200 if item successfully removed.', () => {
+
+    return superagent.delete(`${apiURL}/${user._id}`)
```

Comment:
This fails because user is undefined, though I think you know that, but please try not to submit tests that you know fail.